src/SOS/SOS.Hosting/RuntimeWrapper.cs

```diff
@@ -124,34 +122,26 @@ void IDisposable.Dispose()
         protected override void Destroy()
         {
             Trace.TraceInformation("RuntimeWrapper.Destroy");
-            _onFlushEvent.Dispose();
-            Flush();
-            if (_dacHandle != IntPtr.Zero)
-            {
-                DataTarget.PlatformFunctions.FreeLibrary(_dacHandle);
-                _dacHandle = IntPtr.Zero;
-            }
-            if (_dbiHandle != IntPtr.Zero)
-            {
-                DataTarget.PlatformFunctions.FreeLibrary(_dbiHandle);
-                _dbiHandle = IntPtr.Zero;
-            }
-        }
-
-        private void Flush()
-        {
-            // TODO: there is a better way to flush _corDebugProcess with ICorDebugProcess4::ProcessStateChanged(FLUSH_ALL)
             if (_corDebugProcess != IntPtr.Zero)
             {
                 ComWrapper.ReleaseWithCheck(_corDebugProcess);
                 _corDebugProcess = IntPtr.Zero;
             }
-            // TODO: there is a better way to flush _clrDataProcess with ICLRDataProcess::Flush()
             if (_clrDataProcess != IntPtr.Zero)
             {
                 ComWrapper.ReleaseWithCheck(_clrDataProcess);
                 _clrDataProcess = IntPtr.Zero;
             }
+            if (_dacHandle != IntPtr.Zero)
+            {
+                DataTarget.PlatformFunctions.FreeLibrary(_dacHandle);
```

Comment:
While not interesting in this context, you might take a look at `RefCountedFreeLibrary` in ClrMD.  It gives a way to manually addref/release a library handle, while also ensuring we don't forget and let it go out of scope.  (Not for this checkin, I just mean in general.)

Reply:
I'll keep that in mind.  For this code, we can reason that there will only be one "reference" at anytime to these handles if non-zero.